03_Modules/EVDriver/src/module/api.ts

```diff
@@ -140,12 +143,76 @@ export class EVDriverModuleApi
     CallAction.RequestStartTransaction,
     RequestStartTransactionRequestSchema,
   )
-  requestStartTransaction(
+  async requestStartTransaction(
     identifier: string,
     tenantId: string,
     request: RequestStartTransactionRequest,
     callbackUrl?: string,
   ): Promise<IMessageConfirmation> {
+    if (request.chargingProfile) {
+      const chargingProfile = request.chargingProfile;
+      // Ocpp 2.0.1 Part 2 K05.FR.02
+      if (
+        chargingProfile.chargingProfilePurpose !==
+        ChargingProfilePurposeEnumType.TxProfile
+      ) {
+        return {
+          success: false,
+          payload:
+            'The Purpose of the ChargingProfile SHALL always be TxProfile.',
+        };
+      }
+
+      // Ocpp 2.0.1 Part 2 K05 Description 8 Remarks
+      if (chargingProfile.transactionId) {
+        chargingProfile.transactionId = undefined;
+        this._logger.error(
```

Comment:
very minor semantic thing: should this log be a warn instead of error since the code logic proceeds without actually throwing or anything?